### PR TITLE
fix: add support for eslint 9

### PR DESCRIPTION
**Pull request by reggi**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,44 +1,28 @@
-module.exports = {
-  extends: [
-    'eslint:recommended',
-  ],
-  parserOptions: {
+const globals = require("globals")
+const js = require("@eslint/js")
+const pluginNode = require("eslint-plugin-node")
```
Comment (voxpelli) [inline lib/index.js:3]: Use `eslint-plugin-n` instead, its the currently supported fork and maintained by us in the official `eslint-community`: https://github.com/eslint-community/eslint-plugin-n

File: lib/index.js [hunk 2]
```diff
@@ -1,44 +1,28 @@
...
+const pluginImport = require("eslint-plugin-import")
```
Comment (voxpelli) [inline lib/index.js:4]: Does not yet support ESLint 9: https://github.com/import-js/eslint-plugin-import/issues/2948

See also: https://github.com/neostandard/neostandard/issues/15

File: lib/index.js [hunk 3]
```diff
@@ -1,44 +1,28 @@
...
+const pluginPromise = require("eslint-plugin-promise")
```
Comment (voxpelli) [inline lib/index.js:5]: Does not yet support ESLint 9: https://github.com/eslint-community/eslint-plugin-promise/issues/449 We will get to it within the `eslint-community`

See also: https://github.com/neostandard/neostandard/issues/14

File: lib/index.js [hunk 4]
```diff
@@ -230,4 +214,27 @@ module.exports = {
...
+}, {
+  files: ['lib/**', 'bin/**'],
+  rules: {
+    'import/no-extraneous-dependencies': ['error', { devDependencies: false }],
+  },
+}, {
+  files: ['lib/**'],
+  rules: {
+    'no-console': 'error',
+  },
```
Comment (voxpelli) [inline lib/index.js:233]: Add descriptive names to these to more easily identify them in eg. the config inspector, eg:

```suggestion
{
  name: 'Test adaptions',
  files: ['test/**'],
  rules: {
    'no-global-assign': 'off',
    'no-extend-native': 'off',
  },
}, {
  name: 'Shipping code adaptions',
  files: ['lib/**', 'bin/**'],
  rules: {
    'import/no-extraneous-dependencies': ['error', { devDependencies: false }],
  },
}, {
  name: 'Shipping code adaptions, non-cli',
  files: ['lib/**'],
  rules: {
    'no-console': 'error',
  },
```

File: lib/index.js [hunk 5]
```diff
@@ -1,44 +1,28 @@
...
+const pluginPromise = require("eslint-plugin-promise")
+const { fixupPluginRules } = require("@eslint/compat")
+
+const npmCliConfig = [{
+  languageOptions: {
     ecmaVersion: 2022,
-    ecmaFeatures: {},
     sourceType: 'script',
+    globals: {
+      ...globals.browser,
```
Comment (voxpelli) [inline lib/index.js:13]: The previous `env` was `es6` + `node`, not `browser` + `node`

**Review by voxpelli — CHANGES_REQUESTED** [review]
Also consider basing this config on https://github.com/neostandard/neostandard instead and help out getting a good base config for the ecosystem at large. (`neostandard` is a maintained open governance fork of the `standard` project)

**Comment by reggi** [comment]
@voxpelli thanks for your contributions here, I was editing the `node_modules` version pushing to get it to work and didn't want to loose my changes so I parked them here (should have been a draft pr) thanks for looking at this!

**Pull request closed without merging**